src/main/java/net/dries007/tfc/CommonEventHandler.java

```diff
@@ -565,6 +566,16 @@ public static void onLivingSpawnEvent(LivingSpawnEvent.CheckSpawn event)
         }
     }
 
+    @SubscribeEvent
+    public static void onEntityJoinWorldEvent(EntityJoinWorldEvent event)
+    {
+        // Prevent vanilla animals (that have a TFC counterpart) from mob spawners / egg throws / other mod mechanics
+        if (ConfigTFC.GENERAL.forceNoVanillaAnimals && Helpers.isVanillaAnimal(event.getEntity()))
+        {
+            event.setCanceled(true);
+        }
```

Comment:
Not sure if we should spawn TFC mobs instead. The implications of doing so is:
- Mob spawners would spawn our mobs
- Using vanilla spawn eggs also would spawn our mobs
- Any other mod trying to spawn vanilla mobs would spawn our mobs too (and if said mod would try to do something like Class Cast to `EntityChicken` without checking it's instance, BOOM crash)

At least canceling the event would let the other mod know that the spawning is prevented.

Reply:
I very much doubt that would be an issue. We're not trying to intercept `newInstance`, we're trying to intercept as late as possible, when the entity is actually getting added to the world (`spawnEntity`) More importantly, that method does not return the entity that was actually spawned - unless a mod spawned an entity, then queried the world for the exact entity which they already have a reference to (because they constructed it), *and* then did that unsafely, there would be problems.

As for mob spawners (although what spawners?) or vanilla eggs, I think this is fine, predicate that it's configurable (which it already is)

The effect of spawning a different entity is just as bad as not spawning one - a `ClassCastException` is replaced with a `NullPointerException`